lighthouse-core/gather/driver.js

```diff
@@ -267,11 +267,20 @@ class Driver {
   getAppManifest() {
     return this.sendCommand('Page.getAppManifest')
       .then(response => {
-        // We're not reading `response.errors` however it may contain critical and noncritical
-        // errors from Blink's manifest parser:
-        //   https://chromedevtools.github.io/debugger-protocol-viewer/tot/Page/#type-AppManifestError
+        // Handle JSON syntax errors and manifest correctness warnings
+        if (response.errors) {
+          // Possible issues: https://cs.chromium.org/search/?q=f:manifest_parser.cc+AddErrorInfo&type=cs
+          response.errors.forEach(error => {
+            log.warn('driver', 'Manifest error', error.message);
+          });
+          // Possible critical issues: invalid JSON (location reported) or manifest root is an array
+          const criticalError = response.errors.find(error => error.critical);
+          if (criticalError) {
+            throw new Error(`Manifest error: ${criticalError.message}`);
```

Comment:
We really don't want errors like these reported to Sentry, but this does seem like a valid thing to display as a fatal error for an audit, I guess we should bring back `ExpectedError`?